components/ui-api-layer/docs/proposals/exceed-resource-quota.md

```diff
@@ -0,0 +1,171 @@
+
+>**NOTE** The value of `.spec.strategy.rollingUpdate.maxUnavailable` can be expressed as a percentage or a number. You can find more information [here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#max-unavailable)
+
+If the desired number of replicas is not reached, then check if any ResourceQuota is not blocking the ReplicaSet from progressing.
+To achieve that implement logic which calculates if the ReplicaSet has enough memory to create the next replica.
+To calculate how much resources does the ReplicaSet needs to progress you must sum up the resource usage of all containers in the replica Pod. You also must calculate the `difference` from `.spec.hard` and `.status.used`. It will give you the total amount of available resources to use in the given Namespace.
+Loop through ResourceQuota list and check if any ResourceQuota is not blocking the ReplicaSet from progress by comparing available resources left in the Environment and the required resources to create next replica.
+If there is no enough resources to create the next replica and the number of desired replicas is not reached, then you returns a ResourceQuotaStatus with a flag set to true.
+
+>**NOTE** The Kubeless functions will work with this solution, because they use the Deployment and the ReplicaSet underneath.
+
+### Check the StatefulSets
+
+List StatefulSets and loop through them.
+For each StatefulSet which number of replicas is lower then expected, you must implement logic which checks if any ResourceQuota is not blocking the StatefulSet from progressing.
+To achieve that implement logic which calculates if the StatefulSet has enough memory to create the next replica.
+To calculate how much resources does the StatefulSet needs to progress you must sum up the resource usage of all containers in the replica Pod. You also must calculate the `difference` from `.spec.hard` and `.status.used`. It will give you the total amount of available resources to use in the given Namespace.
+
+Loop through ResourceQuota list and check if any ResourceQuota is not blocking the StatefulSet from progress by comparing available resources left in the Environment and the required resources to create the next replica.
+If there is no enough resources to create the next replica and the number of replicas is not reached, then you returns a ResourceQuotaStatus with a flag set to true.
+
+### End of checking
+
+When all checks passed and the resources usage in the given Environment did not exceed the ResourceQuota, return the ResourceQuotaStatus with a flag set to false. 
+
+[Here](https://github.com/dtaylor113/origin-web-console/blob/master/app/scripts/services/quota.js#L86) you can find the example implementation of the checking logic in UI.
+
+
+## Rejected solutions
+
+These are the solutions rejected by the Kyma team.
+
+### Percentage usage of the ResourceQuota
```

Comment:
You can make it H2 and add a sentence explaining that this solution is not used but may be implemented in the future.

Reply:
Ok, done